### PR TITLE
[GPU] Minor refactoring

### PR DESCRIPTION
**Pull request by yeonbok**
### Details:
 - Refactor according to the comments in PR25449

### Tickets:
 - *ticket-id*

### CODE REVIEW

File: src/plugins/intel_gpu/include/intel_gpu/runtime/layout.hpp [hunk 1]
```diff
@@ -307,7 +311,7 @@ struct layout {
     }
 
     /// Number of elements to be stored in this memory layout
-    size_t count() const;
+    size_t count(bool include_padding = false) const;
```
Comment (vladimir-paramuzov) [inline src/plugins/intel_gpu/include/intel_gpu/runtime/layout.hpp:314]: I suggest having a separate method for that. Maybe `padded_count` ?

Comment (yeonbok) [reply src/plugins/intel_gpu/include/intel_gpu/runtime/layout.hpp]: Updated

File: src/plugins/intel_gpu/include/intel_gpu/runtime/layout.hpp [hunk 2]
```diff
@@ -288,6 +288,15 @@ struct layout {
         return *this;
     }
 
+    layout clone_with_other_shape(const ov::PartialShape& new_shape) {
+        return layout(new_shape, this->data_type, this->format, this->data_padding);
+    }
+
+    layout clone_with_other_shape(const ov::Shape& new_shape) {
+        return layout(ov::PartialShape(new_shape), this->data_type, this->format, this->data_padding);
```
Comment (vladimir-paramuzov) [inline src/plugins/intel_gpu/include/intel_gpu/runtime/layout.hpp:296]:
```suggestion
        return clone_with_other_shape(ov::PartialShape(new_shape));
```

File: src/plugins/intel_gpu/include/intel_gpu/runtime/layout.hpp [hunk 3]
```diff
@@ -306,9 +315,12 @@ struct layout {
         return (lhs.data_padding < rhs.data_padding);
     }
 
-    /// Number of elements to be stored in this memory layout
+    /// Number of elements to be stored in this layout
     size_t count() const;
 
+    /// Number of elements to be stored in this memory layout including padding
+    size_t padded_count() const;
```
Comment (vladimir-paramuzov) [inline src/plugins/intel_gpu/include/intel_gpu/runtime/layout.hpp:322]: btw, looks like this method duplicates `get_linear_size` functionality. Maybe we can use latter one everywhere?

File: src/plugins/intel_gpu/src/graph/primitive_inst.cpp [hunk 4]
```diff
@@ -408,7 +408,6 @@ void primitive_inst::update_shape() {
 
     if (has_runtime_deps) {
         OV_ITT_SCOPED_TASK(ov::intel_gpu::itt::domains::intel_gpu_plugin, openvino::itt::handle("update_shape_sync: " + id()));
-        GPU_DEBUG_TRACE_DETAIL << "runtime synchronization for " << id() << " shape inference\n";
```
Comment (vladimir-paramuzov) [inline src/plugins/intel_gpu/src/graph/primitive_inst.cpp:411]: Is it intended?

File: src/plugins/intel_gpu/src/graph/primitive_inst.cpp [hunk 5]
```diff
@@ -684,13 +683,14 @@ event::ptr primitive_inst::realloc_if_needed() {
             prealloc_shape[seq_axis] += tmp_prealloc_count;
             required_buffer_size = std::accumulate(prealloc_shape.begin(), prealloc_shape.end(), size_t(1), std::multiplies<size_t>());
         } else {
-            required_buffer_size = (updated_layouts[i].get_buffer_size().count());
+            required_buffer_size = (updated_layouts[i].padded_count());
         }
         if (required_buffer_size * 10 < _max_output_layout_count[i]) {
             reclaim = true;
         }
+        reclaim = false;
```
Comment (vladimir-paramuzov) [inline src/plugins/intel_gpu/src/graph/primitive_inst.cpp:691]: Looks like some experimental change. Is it intended?

File: src/plugins/intel_gpu/include/intel_gpu/runtime/layout.hpp (hunk 2 above)
Comment (yeonbok) [reply src/plugins/intel_gpu/include/intel_gpu/runtime/layout.hpp]: Updated

File: src/plugins/intel_gpu/src/graph/primitive_inst.cpp (hunk 4 above)
Comment (yeonbok) [reply src/plugins/intel_gpu/src/graph/primitive_inst.cpp]: oh no, reverted

File: src/plugins/intel_gpu/src/graph/primitive_inst.cpp (hunk 5 above)
Comment (yeonbok) [reply src/plugins/intel_gpu/src/graph/primitive_inst.cpp]: oh no, reverted

File: src/plugins/intel_gpu/include/intel_gpu/runtime/layout.hpp (hunk 3 above)
Comment (yeonbok) [reply src/plugins/intel_gpu/include/intel_gpu/runtime/layout.hpp]: Updated

**Review by vladimir-paramuzov — APPROVED** [review]
LGTM

**Pull request merged**